module/vision/BallDetector/src/BallDetector.cpp

```diff
@@ -233,62 +225,57 @@ namespace module::vision {
-                            log<NUClear::DEBUG>(fmt::format("Camera {}", balls->id));
-                            log<NUClear::DEBUG>(fmt::format("radius {}", b.cone.radius));
-                            log<NUClear::DEBUG>(fmt::format("Axis {}", b.cone.axis.transpose()));
                             log<NUClear::DEBUG>(
-                                fmt::format("Distance {} - rBCc {}", distance, b.measurements.back().rBCc.transpose()));
-                            log<NUClear::DEBUG>(fmt::format("screen_angular {} - angular_size {}",
-                                                            b.screen_angular.transpose(),
-                                                            b.angular_size.transpose()));
-                            log<NUClear::DEBUG>(fmt::format("Projection Distance {}", projection_distance));
-                            log<NUClear::DEBUG>(fmt::format("Distance Throwout {}",
-                                                            std::abs(projection_distance - distance) / max_distance));
-                            log<NUClear::DEBUG>("**************************************************");
+                                fmt::format("Ball discarded: Distance to ball greater than field length: distance = "
+                                            "{}, field length = {}",
+                                            distance,
+                                            field.dimensions.field_length));
+                            log<NUClear::DEBUG>("--------------------------------------------------");
+
+                            b.colour = keep ? message::conversion::math::fvec4(1.0f, 0.0f, 1.0f, 1.0f) : b.colour;
+                            keep     = false;
                         }
 
-                        if (config.debug || keep) {
+                        log<NUClear::DEBUG>(fmt::format("Camera {}", balls->id));
+                        log<NUClear::DEBUG>(fmt::format("radius {}", b.cone.radius));
+                        log<NUClear::DEBUG>(fmt::format("Axis {}", b.cone.axis.transpose()));
+                        log<NUClear::DEBUG>(
+                            fmt::format("Distance {} - srBCc {}", distance, b.measurements.back().srBCc.transpose()));
+                        log<NUClear::DEBUG>(fmt::format("screen_angular {} - angular_size {}",
+                                                        b.screen_angular.transpose(),
+                                                        b.angular_size.transpose()));
+                        log<NUClear::DEBUG>(fmt::format("Projection Distance {}", projection_distance));
+                        log<NUClear::DEBUG>(fmt::format("Distance Throwout {}",
+                                                        std::abs(projection_distance - distance) / max_distance));
+                        log<NUClear::DEBUG>("**************************************************");
+
+                        if (log_level <= NUClear::DEBUG || keep) {
```

Comment:
Is this correct? 

Why are balls only added if we are log level debug?

Reply:
Balls are added if log level debug so we emit them and NUsight can draw them. This does raise a potential issue that Ball Localisation need to reject if they are non "thrownout" balls